src/gui/sharemanager.cpp

```diff
@@ -259,6 +265,14 @@ void LinkShare::setExpireDate(const QDate &date)
     job->setExpireDate(getId(), date);
 }
 
+void LinkShare::setLabel(const QString &label)
+{
+    auto *job = new OcsShareJob(_account);
+    connect(job, &OcsShareJob::shareJobFinished, this, &LinkShare::slotLabelSet);
+    connect(job, &OcsJob::ocsError, this, &LinkShare::slotOcsError);
```

Comment:
@camilasan A proposition for small refactoring. Since this is the 4th method with repeating code (besides the `job->setLabel(getId(), label);` part). We could as well move those to a separate method that would return a pointer to `OcsShareJob` that is already connected to slots. Then, just do `job->setLabel(getId(), label);` or whatever action applies to each of `void LinkShare::set` methods.

Reply:
I did what you said and a bit more. Please check if it makes sense at all.